elixir/apps/domain/lib/domain/component_versions.ex

```diff
@@ -0,0 +1,68 @@
+defmodule Domain.ComponentVersions do
+  alias Domain.ComponentVersions
+  use Supervisor
+  require Logger
+
+  def start_link(opts) do
+    Supervisor.start_link(__MODULE__, opts, name: __MODULE__.Supervisor)
+  end
+
+  @impl true
+  def init(_opts) do
+    pool_opts = Domain.Config.fetch_env!(:domain, :http_client_ssl_opts)
+
+    children = [
+      {Finch, name: __MODULE__.Finch, pools: %{default: pool_opts}},
+      ComponentVersions.Instance
+    ]
+
+    Supervisor.init(children, strategy: :rest_for_one)
+  end
+
+  def gateway_version do
+    ComponentVersions.Instance.component_version(:gateway)
+  end
+
+  def fetch_versions do
+    config = fetch_config!()
+    releases_url = Keyword.fetch!(config, :firezone_releases_url)
+    from_url? = Keyword.fetch!(config, :from_url)
```

Comment:
Is there any reason we have this config? Can't find it's usage

Reply:
Found it. WDYT about renaming it to something like `fetch_from_url?`?

Reply:
👍 Sounds good to me.  This has been updated.